docker/docker_image.go

```diff
@@ -18,8 +18,8 @@ type Image struct {
 
 // NewDockerImage returns a new Image interface type after setting up
 // a client to the registry hosting the given image.
-func NewDockerImage(img, certPath string, tlsVerify bool) (types.Image, error) {
-	s, err := newDockerImageSource(img, certPath, tlsVerify)
+func NewDockerImage(img string, dc Client) (types.Image, error) {
+	s, err := newDockerImageSource(img, dc)
```

Comment:
Why should the client care about our internal testing interface?

Can’t we do
```go
func newDockerImage(img string, dc Client) … {}

func NewDockerImage(img, certPath string, tlsVerify bool) (…) {
    client, err := NewClient(img, certPath, tlsVerify)
    if err != nil {…}
    return newDockerImage(img, client)
}
```
That would leave `NewDockerImage` untested, sure. Or we can at least do a smoke test.